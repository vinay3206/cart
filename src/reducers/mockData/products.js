const imageUrl = 'https://4.bp.blogspot.com/-aI_KMIohZHQ/VN_63b-lsQI/AAAAAAAAZd4/6sifxckbEgQ/s1600/F%26N%2BNutriTea%2BBarley%2BDrink_1L.jpg';

const products = [
  {
    id: 120,
    capacity: 1,
    capacityUnit: 'L',
    title: 'Nutri Well Barly',
    price: 2.25,
    currency: '$',
    imageUrl,
    brand: {
      id: 1,
      label: 'Mangolia',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
  {
    id: 121,
    capacity: 1,
    capacityUnit: 'L',
    title: 'Nutri Well Lemon Grass with Ginger',
    price: 2.35,
    currency: '$',
    imageUrl,
    brand: {
      id: 2,
      label: 'Meiji',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
  {
    id: 122,
    capacity: 500,
    capacityUnit: 'ml',
    title: 'Nutri Well Water Chestnut and Sugar cane',
    price: 1.25,
    currency: '$',
    imageUrl,
    brand: {
      id: 2,
      label: 'Meiji',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
  {
    id: 123,
    capacity: 1,
    capacityUnit: 'L',
    title: 'Nutri Well Water cherysenthamums with Wolfberry',
    price: 2.25,
    currency: '$',
    imageUrl,
    brand: {
      id: 1,
      label: 'Mangolia',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
  {
    id: 124,
    capacity: 1,
    capacityUnit: 'L',
    title: 'Nutri Uji Cha Yuzu Green Tea',
    price: 5.05,
    currency: '$',
    imageUrl,
    brand: {
      id: 1,
      label: 'Mangolia',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
  {
    id: 125,
    capacity: 1,
    capacityUnit: 'L',
    title: 'Marigold Uji Cha Momo Green Tea',
    price: 7.25,
    currency: '$',
    imageUrl,
    brand: {
      id: 2,
      label: 'Meiji',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
];

export default products;
