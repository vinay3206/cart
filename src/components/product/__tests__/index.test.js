import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Product from '../index';
import products from '../../../reducers/mockData/products';

const product = products[0];

it('renders correctly when count is 0 and add to cart is enabled', () => {
  const component = <Product
    product={product}
    onAddToCart={() => {}}
    count={0}
    isAddToCartDisabled={false}
  />;
  const tree = renderer
    .create(
      <MemoryRouter>{component}</MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when count is 1 and add to cart is disabled', () => {
  const component = <Product
    product={product}
    onAddToCart={() => {}}
    count={1}
    isAddToCartDisabled
  />;
  const tree = renderer
    .create(
      <MemoryRouter>{component}</MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
