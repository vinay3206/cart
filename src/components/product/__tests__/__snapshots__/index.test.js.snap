// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when count is 0 and add to cart is enabled 1`] = `
<div
  className="d-inline-flex"
>
  <a
    className="product-card"
    href="/detail/120"
    onClick={[Function]}
  >
    <img
      alt={120}
      src="https://4.bp.blogspot.com/-aI_KMIohZHQ/VN_63b-lsQI/AAAAAAAAZd4/6sifxckbEgQ/s1600/F%26N%2BNutriTea%2BBarley%2BDrink_1L.jpg"
      style={
        Object {
          "height": "200px",
        }
      }
    />
    <h4
      className="product-card-title"
    >
      Nutri Well Barly
       
       
       
      Mangolia
    </h4>
    <div>
      
    </div>
    <div>
      $
      2.25
    </div>
    <div>
      <button
        className="btn-flat btn-red"
        disabled={false}
        onClick={[Function]}
      >
        Add to Cart
      </button>
    </div>
  </a>
  ;
</div>
`;

exports[`renders correctly when count is 1 and add to cart is disabled 1`] = `
<div
  className="d-inline-flex"
>
  <a
    className="product-card"
    href="/detail/120"
    onClick={[Function]}
  >
    <img
      alt={120}
      src="https://4.bp.blogspot.com/-aI_KMIohZHQ/VN_63b-lsQI/AAAAAAAAZd4/6sifxckbEgQ/s1600/F%26N%2BNutriTea%2BBarley%2BDrink_1L.jpg"
      style={
        Object {
          "height": "200px",
        }
      }
    />
    <h4
      className="product-card-title"
    >
      Nutri Well Barly
       
       
       
      Mangolia
    </h4>
    <div>
       Qunatity: 1
    </div>
    <div>
      $
      2.25
    </div>
    <div>
      <button
        className="btn-flat btn-red"
        disabled={true}
        onClick={[Function]}
      >
        Add to Cart
      </button>
    </div>
  </a>
  ;
</div>
`;
